tests/unittests/InterpreterTest.cpp

```diff
@@ -898,12 +898,92 @@ TEST(Interpreter, nonLinearClassifier) {
   }
 }
 
+/// Generate images in two classes. A "line" is labeled as 0
+/// and a "cross" is labeled as 1
+static void generateImageData(Tensor &images, Tensor &labels) {
+  auto L = labels.getHandle<size_t>();
+  auto image = images.getHandle<>();
+  unsigned numSamples = images.dims()[0];
+  images.zero();
+
+  for (size_t i = 0; i < numSamples; i++) {
+    bool isLine = i % 2 == 0;
+    L.at({i, 0}) = isLine ? 0 : 1;
+    size_t target = nextRandInt(1, 6);
+    if (isLine) {
+      for (size_t y = 0; y < 8; y++)
+        image.at({i, target, y, 0u}) = 1;
+    } else {
+      for (size_t pos = 0; pos < 8; pos++) {
+        image.at({i, pos, target, 0u}) = 1;
+        image.at({i, target, pos, 0u}) = 1;
+      }
+    }
+  }
+}
+
+/// Test converlutional layer. Use a simple convnet
```

Comment:
typo.

Reply:
Also, period at the end of the sentence.